### DISCUSSION TITLE
Is it possible to generate a terminal-based summary for peak memory usage table ?

### DISCUSSION THREAD

**Discussion by tvalentyn** [bloomberg/memray#479] [Q&A]
My current workflow is:

1) convert the bin output into html table via `memray table`
2) open the table in the browser 
3) sort the rows by the size in decreasing order (two clicks).

Is possible to get the same info from the terminal? If not, would it make sense to file a feature request? 

Thanks!

**Comment by pablogsal** [comment]
Have you seen https://bloomberg.github.io/memray/summary.html? Does this work for your use case?

**Reply by tvalentyn** [reply]
Thanks, it looks like exactly what I need. 

I just realized that I tried using `summary` in the very beginning and forgot about it. It didn't work for me because it trimmed the  filenames too much, so the output was not usable.

**Reply by tvalentyn** [reply]
Also, by comparing the output of `summary`, it looks to be different from content produced by `table`. Also, one difference, summary doesn't support `--leak` mode.

**Reply by pablogsal** [reply]
Yep, they show different things. Table shows all allocations individually and summary focus on locations.

We are adding the --leaks soon to table and tree reporters.

**Reply by tvalentyn** [reply]
Thank you!

> it trimmed the filenames too much, so the output was not usable.

Trimming from the beginning, instead of the end, would work better in this case as file name is more meaningful than the preceding path.

**Reply by tvalentyn** [reply]
I took a look once again at `summary` output, in a wider terminal, and the output doesn't show the leak that I am able to see in the `table` output, so I think as is the `summary` command doesn't work for my usecase.
In anycase, memray is beyond awesome and I owe to post a success story once I finish debugging.

**Reply by tvalentyn** [reply]
In one example, `summary` shows a table with only 6 rows, while the `table` output has 10k rows.

**Reply by pablogsal** [reply]
Right, notice that these tools show different information: `memray table` shows you every individual allocation (one by one) that is in the high watermark or the leak set, while `memray summary` shows you aggregated allocations by location. In `memray summary` the functions themselves are shown and not allocations. This means that if a function has a call tree of `foo->bar->baz` and `baz` allocates, we will show 3 rows, even if there is only one allocation.  Another thing to consider is that we only show a limited number of rows in `memray summary` (that can be controlled via a command line parameter). 

In any case, if you can give as a simple reproducer to take a look the difference between `table` and `summary` in your case, we can help you explain why one has 10k rows and the other 6.

**Reply by pablogsal** [reply]
Btw, in general, you may find `memray flamegraph` the most informative reporter, because this shows the same information in a more graphical format.
